File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day11HexEdTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day11HexEdTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-input.txt");

        assertEquals(877, dp.getPart1Answer().intValue());
        assertEquals(1622, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-example#1.txt");

        assertEquals(3, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-example#2.txt");

        assertEquals(0, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample3() {
        DailyProblem<Integer, Integer> dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-example#3.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample4() {
        DailyProblem<Integer, Integer> dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-example#4.txt");

        assertEquals(3, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day11HexEdTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day11HexEdTest  extends FlatSpec with Matchers {
  "2017 Day 11 - Input" should "calculate answers" in {
    val dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day11HexEd("/twenty_seventeen/Day11-HexEd-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
